app/src/main/java/com/arygm/quickfix/ui/elements/QuiickFixServicesChoosing.kt

```diff
@@ -0,0 +1,168 @@
+package com.arygm.quickfix.ui.elements
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.Divider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.MaterialTheme.colorScheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import com.arygm.quickfix.ui.theme.QuickFixTheme
+
+/**
+ * A composable function that displays a modal sheet for selecting a service type. It includes a
+ * list of service options, an "Apply" button, and a "Reset" option. The sheet uses relative
+ * dimensions for responsiveness across various screen sizes.
+ *
+ * @param showModalBottomSheet Controls the visibility of the modal sheet.
+ * @param serviceTypes List of service types to display as selectable options.
+ * @param selectedService The currently selected service.
+ * @param onServiceSelect Callback triggered when a service option is selected.
+ * @param onApplyClick Callback triggered when the "Apply" button is clicked.
+ * @param onResetClick Callback triggered when the "Reset" text is clicked.
+ * @param onDismissRequest Callback triggered when the modal sheet is dismissed.
+ */
+@Composable
+fun ChooseServiceTypeSheet(
+    showModalBottomSheet: Boolean,
+    serviceTypes: List<String>,
+    selectedService: String,
+    onServiceSelect: (String) -> Unit,
+    onApplyClick: () -> Unit,
+    onResetClick: () -> Unit,
+    onDismissRequest: () -> Unit
+) {
+  if (showModalBottomSheet) {
+    Dialog(
+        onDismissRequest = onDismissRequest,
+        properties = DialogProperties(usePlatformDefaultWidth = false)) {
+          BoxWithConstraints {
+            val paddingHorizontal = maxWidth * 0.04f // Relative horizontal padding (4% of width)
+            val verticalSpacing = maxHeight * 0.015f // Relative vertical spacing (1.5% of height)
+            val cornerRadius = maxWidth * 0.05f // Rounded corner radius (5% of width)
+            val buttonPaddingHorizontal = maxWidth * 0.04f // Button padding (4% of width)
```

Comment:
Nice way of adding relative dimensions, but i think for a future implementation it would be better to use weights instead of BoxWithConstraints.

Reply:
Thanks for your comment! I'll take a look at the documentation and try using weights as you suggested. It's a great idea for a future implementation. Thanks again for your feedback!